framework/src/modules/interoperability/base_interoperability_store.ts

```diff
@@ -241,65 +244,72 @@ export abstract class BaseInteroperabilityStore {
 		return terminatedOutboxSubstore.get(this.context, chainID);
 	}
 
-	public async createTerminatedStateAccount(chainID: Buffer, stateRoot?: Buffer): Promise<boolean> {
-		const chainSubstore = this.stores.get(ChainAccountStore);
-		const isExist = await this.chainAccountExist(chainID);
+	public async createTerminatedStateAccount(
+		context: CreateTerminatedStateAccountContext,
```

Comment:
why adding context here? we can use `this.context` instead within the methods of store class

Reply:
Because we need to pass in `eventQueue` to the events